awslabs/amazon-timestream-connector-prometheus
# Remove Multi-destination Support

### Summary

Remove Multi-destination support from documentation and implementation.

### Description
Using labels to determine the ingestion and query destination is a source of confusion and therefore being removed from the documentation and implementation for the release of the Prometheus Connector. All ingestion and query destinations will now go to the database and table defined with the `default-database` and `default-table` variables for the Prometheus Connector. Multi-destination ingestion is not supported once this PR is merged. There was a bug with the SAM template where the lambda function could not create a log group which has been resolved. The example database and table have been updated in all documentation examples with the more appropriate names `PrometheusDatabase`, and `PrometheusTable`. 

### Additional Changes
- Add documentation for how to execute TLS tests
- Add documentation for how to generate new TLS certificates for testing
- Generated new TLS certificates for testing
- Add documentation for how to execute correctness tests
- Fix invalid prometheus configuration template for TLS tests
- Added correctness tests to the contribution guide
- Release workflow was fixed as the `hub` command not installed by default with the GH action runner
- Updated documentation to use `buildx` command when building the docker image to avoid any ARM compatibility issues

### Verification

- Test remote write and read queries running connector locally
  - _execute with the following command_ ` ./timestream-prometheus-connector --default-database=PrometheusDatabase --default-table=PrometheusMetricsTable  --region us-east-1`
  - [x] Verified ingestion is successful with no database/table configured in Prometheus configuration
    - _Verified data is being collected in timestream by querying the table with the following query and seeing the records increase_ `SELECT count(*) FROM 'PrometheusDatabase'.'PrometheusMetricsTable'`
  - [x] Verified ingestion fails with database/table labels configured in Prometheus configuration
    - _Add `write_relabel_configs` section to `prometheus.yml` configuration with labels `PrometheusDabaseLabel` and `PrometheusTableLabel`_
  - [x] Verified remote_read queries are successful with no database/table labels in the PromQL query
    - _verified with the following command, note the 99d is used to ensure a remote_read is triggered_ `scrape_duration_seconds{}[99d]`
  - [x] Verified remote_read queries fail with database/table labels in the PromQL query
    - _verified with the following command_ `scrape_duration_seconds{PrometheusDatabaseLabel="PrometheusDatabase", PrometheusTableLabel="PrometheusMetricsTable"}[99d]`
- Tested deploying to cloudformation
  - _Deployed by building the deployment artifact and the sam CLI_
    - `python3 ./package.py -v 1.1.0`
    - `mv timestream-prometheus-connector-linux-amd64-1.1.0.zip serverless`
    - `sam deploy -t serverless/template.yml`
  - [x] Verified remote_read queries are successful with no database/table labels in the PromQL query
    - _verified with the following command, note the 99d is used to ensure a remote_read is triggered_ `scrape_duration_seconds{}[99d]`
  - [x] Verified ingestion is successful with no database/table configured in Prometheus configuration
    -  _Verified data is being collected in timestream by querying the table with the following query and seeing the records increase_ `SELECT count(*) FROM 'PrometheusDatabase'.'PrometheusMetricsTable'`
  - [x] Cloudwatch logs are being created
    - _Navigating to the Lambda generated by the CFN template and click on 'View Cloudwatch Logs', then select 'start tailing' and verify logs are collected during ingestion_
- [x] Validate IT tests pass
  - _Verified all tests pass with the following command_ - `go test -v ./integration`
- [x] Validate UT tests pass
  - _Verified all tests pass with the following command_ - `go test -tags=unit -cover ./timestream ./`
- [x] Validate TLS tests pass
  - _Verified all tests pass with the following command_ - `go test ./integration/tls`
- [x] Validate correctness tests pass
  - _Verified all tests pass with the following command_ - `go test ./correctness`

Resolved Issues:
- N/A
